subsys/s11n/Kconfig

```diff
@@ -0,0 +1,87 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig S11N
+	bool
+	default n
+	prompt "Enable serialization system with non-volatile storage"
+	depends on FILE_SYSTEM_NFFS || FCB
```

Comment:
Here you can say CONFIG_FILE_SYSTEM right? Actual file system you can select in the test application. That way you can avoid the dependency on a specific file system.

Reply:
So, Do you mena that if I change this line to
 `depends on (FILE_SYSTEM && FILE_SYSTEM_NFFS) || FCB`
then the line
`zephyr_include_directories($ENV{ZEPHYR_BASE}/ext/fs/nffs/include)`
in  subsys/s11n/src/CMakeLists.txt file will be unnecessary?

Reply:
=> depends on FILE_SYSTEM || FCB

yes, move file system specific details(FILE_SYSTEM_NFFS) to application Cmakefiles/prj.conf.

That way subsystem only depends on FS and not on any particular(NFFS) file system.

Reply:
The problem is that the settings system only works with NFFS as of today I believe

Reply:
As far as i believe, the s11n core does not directly use any of nffs API's or structs. It only depends on FS API's. So the subsystem should build without any issues.

Next, to actually test, as i understand we have the dependency on NFFS (for now). This NFFS specific details can be pushed to test or sample application prj.conf and Cmakefile.txt.

@nvlsianpu Can you confirm if you can do the above change?

Reply:
Yep I think you are right, just verified that. Will push a fix

Reply:
@ramakrishnapallala I just pushed a fix to address your comment. I was mislead by an inclusion of nffs.h, but that has nothing to do with the Kconfig deps.

Reply:
np :)

Reply:
Thanks for the feedback, please approve if happy with the current state.  Note the renaming of the subsytem to `settings`, see the mailing list for discussion.